### PR TITLE
Bugfix: Copy / move file name exists count duplicate

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/7089

## Related Issue

- Fixes https://github.com/owncloud/web/issues/7089

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/src/helpers/resource/copyMove.ts [hunk 1]
```diff
@@ -241,7 +252,9 @@ export const copyMoveResource = async (
   )
   const movedResources = []
 
-  for (const resource of resourcesToMove) {
+  for (let resource of resourcesToMove) {
+    // because javascripts way of cloning is awesome
```
Comment (kulmann) [inline packages/web-app-files/src/helpers/resource/copyMove.ts:256]: please be more precise why a shallow copy is not sufficient 😁

File: packages/web-app-files/src/helpers/resource/copyMove.ts [hunk 2]
```diff
@@ -215,6 +215,14 @@ export const copy = async (
     true
   )
 }
+
+export const resolveFileNameDuplicate = (name, existingFiles, iteration = 1) => {
+  const potentialName = `${name} (${iteration})`
```
Comment (kulmann) [inline packages/web-app-files/src/helpers/resource/copyMove.ts:220]: this only works for folders, not for files, right?

File: packages/web-app-files/tests/unit/helpers/resource/copyMove.spec.ts [hunk 3]
```diff
@@ -23,6 +23,22 @@ describe('copyMove', () => {
       webDavPath: '/target'
     }
   })
+
+  it('should name duplicate file correctly', async () => {
```
Comment (kulmann) [inline packages/web-app-files/tests/unit/helpers/resource/copyMove.spec.ts:27]: please also add test cases with files / file extensions. probably a good idea to make it a table test (look for `it.each` examples in existing unit tests)

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/helpers/resource/copyMove.ts (hunk 2 above)
Comment (lookacat) [reply packages/web-app-files/src/helpers/resource/copyMove.ts:220]: ah yes, right

File: packages/web-app-files/src/store/mutations.js [hunk 4]
```diff
@@ -324,8 +324,11 @@ export default {
 // eslint-disable-next-line camelcase
 function $_upsertResource(state, resource, allowInsert) {
   const files = [...state.files]
-  const index = files.findIndex((r) => r.id === resource.id)
-  const found = index > -1
+  let index = files.findIndex((r) => r.id === resource.id)
+  if(resource.webDavPath && resource.webDavPath.length) {
```
Comment (kulmann) [inline packages/web-app-files/src/store/mutations.js:328]: Please add `index === -1` as an additional condition to the `if` clause. You check that in the body and basically do nothing in that case, so you could as well skip the block entirely, which makes it easier to read.

Comment (kulmann) [reply packages/web-app-files/src/store/mutations.js:328]: is it possible at all the the `webDavPath` is empty?

Comment (lookacat) [reply packages/web-app-files/src/store/mutations.js:328]: yes sadly in some unittests it can happen that webDavPath is empty

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**